### ISSUE TITLE
Node Operations

### ISSUE THREAD

**Issue by scirner22** [kube-rs/kube-rs#706]
Is it possible to perform node operations like add labels, drain, cordon, etc. I'm not seeing it in the Api documentation at all.

**Comment by kazk** [comment]
Add label:
```rust
let nodes: Api<Node> = Api::all(client);
nodes
    .patch(
        "name",
        &PatchParams::default(),
        &Patch::Merge(serde_json::json!({ "metadata": { "labels": { "app": "example" } } })),
    )
    .await?;
```

Cordon should be:
```rust
let nodes: Api<Node> = Api::all(client);
nodes
    .patch(
        "name",
        &PatchParams::default(),
        &Patch::Merge(serde_json::json!({ "spec": { "unschedulable": true } })),
    )
    .await?;
```

Drain should be cordon + evicting all pods in it.

You can use `kubectl` with `--v=9` to find the API requests.

We can probably add convenience methods like in https://github.com/kube-rs/kube-rs/pull/635/files for cordon/uncordon.

**Comment by clux** [comment]
Yeah, even in `client-go` operations like `drain` and `cordon` are not among [its core verbs](https://github.com/kubernetes/client-go/blob/f6ce18ae578c8cca64d14ab9687824d9e1305a67/kubernetes/typed/core/v1/node.go#L42-L56). These "operations" are just `kubectl` helpers: [cordon.go](https://github.com/kubernetes/kubectl/blob/652881798563c00c1895ded6ced819030bfaa4d7/pkg/drain/cordon.go) + [drain.go](https://github.com/kubernetes/kubectl/blob/652881798563c00c1895ded6ced819030bfaa4d7/pkg/drain/drain.go)

I'm ok with having `drain` + `cordon` as utility helpers or examples (as a start). We support the eviction subresource (`Api::evict`) necessary for this. If you would like to try writing a `drainer` example or a utility fn for `Drain` then that would be welcome.

**Comment by clux** [comment]
#762 now adds `Api<Node>::cordon` and `Api<Node>::uncordon` thanks to @ChinYing-Li :tada: 

The only thing left here is an example for `Api<Node>::drain`. As it stands currently herein, this is non-trivial to unit-test / integration test an `Api<Node>::drain` fn since it probably requires a real node, with real pods, to be created and drained. I think it's OK for this to be done in an example for now. But am open to suggestions. kubectl does have some [fairly verbose tests for drain](https://github.com/kubernetes/kubectl/blob/652881798563c00c1895ded6ced819030bfaa4d7/pkg/drain/drain_test.go) - no idea how fast or sensible these are yet.
